Mathlib/Algebra/Module/LocalizedModule/Basic.lean

```diff
@@ -355,7 +352,7 @@ private theorem smul_zero_aux (x : T) : x • (0 : LocalizedModule S M) = 0 := b
 
 private theorem add_smul_aux (x y : T) (p : LocalizedModule S M) :
     (x + y) • p = x • p + y • p := by
-  induction' p with m s
+  cases p with | _ m s => ?_
```

Comment:
This looks like a strictly more indecipherable syntax than what it is being changed from. What is the rationale for this kind of change?

Reply:
That's what @eric-wieser told me to put, blanking out the sole branch's name.

Reply:
Right, but in the case where there is just one introduced variable, you make the line longer and increase functional programming boilerplate. What is the actual benefit over `induction'`?

Reply:
I think the end goal here is to phase out `induction'` entirely, but I think this is a case where today the replacements are all worse; and so perhaps we ought to wait for a better replacement here.